src/Controller/QueryFieldRestController.php

```diff
@@ -30,28 +32,49 @@ final class QueryFieldRestController
     /** @var \eZ\Publish\API\Repository\LocationService */
     private $locationService;
 
+    /** @var \EzSystems\EzPlatformRest\RequestParser */
+    private $requestParser;
+
     public function __construct(
         QueryFieldService $queryFieldService,
         ContentService $contentService,
         ContentTypeService $contentTypeService,
-        LocationService $locationService
+        LocationService $locationService,
+        RequestParser $requestParser
     ) {
         $this->queryFieldService = $queryFieldService;
         $this->contentService = $contentService;
         $this->contentTypeService = $contentTypeService;
         $this->locationService = $locationService;
+        $this->requestParser = $requestParser;
     }
 
     public function getResults(Request $request, $contentId, $versionNumber, $fieldDefinitionIdentifier): RestValues\ContentList
     {
         $offset = (int)$request->query->get('offset', 0);
         $limit = (int)$request->query->get('limit', -1);
 
-        $content = $this->contentService->loadContent($contentId, null, $versionNumber);
-        if ($limit === -1 || !method_exists($this->queryFieldService, 'loadContentItemsSlice')) {
-            $items = $this->queryFieldService->loadContentItems($content, $fieldDefinitionIdentifier);
+        if ($request->query->has('location')) {
+            $locationHrefParts = explode('/', $this->requestParser->parseHref($request->query->get('location'), 'locationPath'));
+            $locationId = array_pop($locationHrefParts);
+            $location = $this->locationService->loadLocation($locationId);
```

Comment:
As far as I can see `$locationId` will be string, so it should be casted to `int` before passing to `$this->locationService->loadLocation()`.

Also maybe we can extract this logic to private, well named method ? 

```php
    private function loadLocationByPath(string $path): Location
    {
        $segments = explode('/', $path);
        if (empty($segments)) {
            throw new \InvalidArgumentException('Empty path provided');
        }

        $locationId = (int)array_pop($segments);

        return $this->locationService->loadLocation($locationId);
    }
```

Reply:
Good idea, done.